opta/meister.py

```diff
@@ -0,0 +1,174 @@
+import atexit
+import inspect
+import os
+import sys
+import textwrap
+from asyncio import Lock, iscoroutinefunction
+from functools import wraps
+from math import ceil, floor, log10
+from time import time as unix_time
+from typing import Any, Callable, Dict, Final, List, Tuple, TypedDict
+from uuid import UUID, uuid4
+
+from colored import attr, fg
+
+
+def round_sig(x: float, sig: int = 3) -> float:
+    return round(x, sig - int(floor(log10(abs(x)))) - 1)
+
+
+class TimedCall(TypedDict):
+    id: UUID
+    function_name: str
+    positional_args: Tuple[Any, ...]
+    keyword_args: Dict[str, Any]
+    start_time: float
+    end_time: float
+    caller_file: str
+    caller_line: int
+
+
+_records: List[TimedCall] = []
+ASYNC_LOCK: Final = Lock()
+REPORT_SIZE: Final = 20
+TIME_THRESHOLD: Final = 0.1
+REPORT_ENABLED: bool = True
```

Comment:
```suggestion
REPORT_ENABLED: bool = False
```

This will avoid measuring functions when running unit tests.